### ISSUE TITLE
Potentially Redundant Code Block & Question

### ISSUE THREAD

**Issue by mstrYoda** [nopSolutions/nopCommerce#3947]
Hi, I saw that code block in the `OrderService`. Is not that foreach loop redundant? What is the logic behind that? If it's only purpose is to sort orders according to parameter orders, is not that slows down the response time considering a list with lots of items?

```
        /// <summary>
        /// Get orders by identifiers
        /// </summary>
        /// <param name="orderIds">Order identifiers</param>
        /// <returns>Order</returns>
        public virtual IList<Order> GetOrdersByIds(int[] orderIds)
        {
            if (orderIds == null || orderIds.Length == 0)
                return new List<Order>();

            var query = from o in _orderRepository.Table
                        where orderIds.Contains(o.Id) && !o.Deleted
                        select o;
            var orders = query.ToList();
            //sort by passed identifiers
            var sortedOrders = new List<Order>();
            foreach (var id in orderIds)
            {
                var order = orders.Find(x => x.Id == id);
                if (order != null)
                    sortedOrders.Add(order);
            }

            return sortedOrders;
        }
```

**Comment by onestar1** [comment]
me too。

**Comment by skoshelev** [comment]
Hi @mstrYoda. This code is needed to sort the result according to the transmitted identifiers. This method allows you to export orders in the selected order, which is very important for many. We don’t think that often there is a need to export a large number of orders, choosing them manually from the list, so we may not worry a lot about the performance of this operation.

Closed #3947
